core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -42,10 +44,15 @@ import scala.concurrent.Future
  *                                     |                      |
  *                                     +----------------------+
  */
-object Scheduler {
+object Scheduler extends LazyLogging {
+
+  val log = Logger.takingImplicit[KvArgs](getClass)
+
   case class MesosConnection(mesosHostName: String)
 
   def connect(mesosHostName: String): Future[(MesosConnection, Flow[SpecEvent, StateEvent, NotUsed])] = {
+    val composedLogger = log.ctx(("mesosHostName", mesosHostName))
+    composedLogger.info(s"Connecting to $mesosHostName")(composedLogger.getCtx)
```

Comment:
I missed some discussion but why is `composedLogger.getCtx` required here? Isn't the context only what we defined above?

Reply:
when calling the [`log.info` method](https://github.com/lightbend/scala-logging/blob/master/src/main/scala/com/typesafe/scalalogging/LoggerTakingImplicit.scala#L44), we have pass an implicit parameter for the context. I tried to do this in couple ways:
1. First, i tried to redefine the `log.info` method, but this was interfering with caller context (Such as the method name and line number from where the logger statement was made). 
2. Then I tried to define the implicit object for context, but did not feel this to be very readable. Also there may be multiple implicits in scope and felt it would not be readable.
After trying above two, i settled for above style of explicitly passing the implicit context.

Reply:
That makes sense to me but Why do I need the first line then? I'd expect

```scala
composedLogger.info(s"Connecting to $mesosHostName")(("mesosHostName", mesosHostName))
```

The extra `withContext` does not make much sense to me.